odo/backends/sparksql.py

```diff
@@ -42,10 +56,29 @@ def jsonlines_to_sparksql(ctx, json, dshape=None, name=None, schema=None,
         schema = dshape_to_schema(dshape.measure
                                   if isrecord(dshape.measure) else dshape)
     srdd = ctx.jsonFile(json.path, schema=schema, samplingRatio=samplingRatio)
-    ctx.registerRDDAsTable(srdd, name or next(_names))
+    ctx.registerDataFrameAsTable(srdd, name or next(_names))
     return srdd
 
 
+@convert.register(list, SparkDataFrame, cost=200.0)
+def sparksql_dataframe_to_list(df, dshape=None, **kwargs):
+    result = df.collect()
+    if (dshape is not None and iscollection(dshape) and
+            not isrecord(dshape.measure)):
+        return list(map(get(0), result))
+    return result
+
+
+@convert.register(base, SparkDataFrame, cost=200.0)
+def spark_df_to_base(df, **kwargs):
+    return df.collect()[0][0]
+
+
+@convert.register(pd.DataFrame, SparkDataFrame, cost=200.0)
+def spark_df_to_pandas_df(df, **kwargs):
+    return df.toPandas()
```

Comment:
Even if it isn't faster it's probably better to give it a lower cost if this edge is more robust.